src/plugins/intel_gpu/src/runtime/debug_configuration.cpp

```diff
@@ -238,16 +244,50 @@ const debug_configuration *debug_configuration::get_instance() {
 #endif
 }
 
-bool debug_configuration::is_dumped_layer(const std::string& layerName, bool is_output) const {
+bool debug_configuration::is_dumped_layer(const std::string& layer_name, bool is_output) const {
 #ifdef GPU_DEBUG_CONFIG
     if (is_output == true && dump_layers_result == 1 &&
-        (layerName.find("constant:") == std::string::npos))
+        (layer_name.find("constant:") == std::string::npos))
         return true;
     if (dump_layers.empty() && dump_layers_result == 0)
         return true;
 
+    auto is_match = [](const std::string& layer_name, const std::string& pattern) -> bool {
+        // Check pattern from exec_graph
+        size_t pos = layer_name.find(':');
+        auto exec_graph_name = layer_name.substr(pos + 1, layer_name.size());
+        if (strcasecmp(exec_graph_name.c_str(), pattern.c_str()) == 0) {
+            return true;
+        }
+        // Check pattern for wildcard letter
+        size_t n = layer_name.size();
+        size_t m = pattern.size();
+        std::vector<std::vector<bool>> lookup_tbl(n + 1, std::vector<bool>(m + 1, false));
+        lookup_tbl[0][0] = true;
+        auto to_upper = [](char elem) -> char {
+            if (elem >= 'a' && elem <= 'z') {
+                return elem - ('a'-'A');
+            }
+            return elem;
+        };
+        for (size_t j = 1; j <= m; ++j)
+            if (pattern[j - 1] == '*')
+                lookup_tbl[0][j] = lookup_tbl[0][j - 1];
+        for (size_t i = 1; i <= n; ++i) {
+            for (size_t j = 1; j <= m; ++j) {
+                if (pattern[j - 1] == '*') {
```

Comment:
how about 
1) replace start to .* 
2) replace ? to .
3) use std::regex_match ?
instead of custom matcher.

Reply:
Good suggestion, I updated to use std::regex_match instead custom matcher. it looks much simpler.